### ISSUE TITLE
Update asset location when checking out to user

### ISSUE THREAD

**Issue by gilesytheking90** [snipe/snipe-it#7560]
**Is your feature request related to a problem? Please describe.**

When checking out an asset to a user, there isn't an option to update the "Current location" of the asset. We have employees working at various sites / locations, so it's useful to know who has what and where it's located. 

Assets display as checked out to a user, but the location remains as the default.

When **checking in** an asset, there is an option to update the location if the item is not being checked in/ returned to the default.

**Describe the solution you'd like**

Add an option to choose/update the location of an asset when checking out to a user. This would update the "Current Location" of the asset. The default location would remain the same.

If no location is selected on checkout, the default location would be used.

**Describe alternatives you've considered**

I currently use the notes section to track / log the location of the asset when being checked out to a user.

**Comment by jacobsen9026** [comment]
I believe this functionality already exists and is working fine on my installation.

As long as the User has a Location assigned in Snipe it should update the location of the asset to match (does on mine). I think it also runs a script nightly to update all assets in case a user's location changes.

**Comment by gilesytheking90** [comment]
I wasn't aware of that functionality but I will definitely test that out. At the moment, I don't have all our employees assigned to a location, mainly because they move around on a day-to-day basis. 

Generally our tools are signed out in the morning and signed back in at night. Updating the location on checkout would be easier then updating each employees location daily.

**Comment by kgortmak** [comment]
Afaik Snipe indeed reports the users location to the asset during checkout. So when the user does not have a location assigned this makes sense. When you check IN the asset you are provided with an option to update the location (in fact there is also an option to update the asset's actual location AND the default location of the asset). I think that when checking OUT the same rule could be used: If you do NOT specify a location, the users location will be used, otherwise the spcified location must be used.

Kornelius

**Comment by SchachtnerTh** [comment]
On our side, people also work in multiple offices. We do have configured a location for a user (that's where his main desk is located), but an asset is not necessarily at that location. It would be good to know where which asset is actually located. (Of course, this is not so easy for assets like notebooks, but TFT displays or heavier devices usually stay where they are...)

I agree with kgortmak: The following would be great:
- User has a (default) location configured
- when checking out an asset do an user, I would like to have the option to fill a field "Location"
- If that field ("Location") is empty, the default location as defined in the user object will be used as the new asset location
- if it is not empty, the location specified there is the new location of the asset.

**Comment by snipe** [comment]
@SchachtnerTh That’s sort of the opposite of the way to works now. *Assets* have a default location, not users. If an asset is unassigned, it stays in its default location. If it is assigned, it will inherit the location (while it’s checked out) to the user it’s assigned to. When it gets checked back in, it will return to its default location.

**Comment by SchachtnerTh** [comment]
I know. And that is a cool feature. We need that, too. But in addition to that, it would be good for us to also be able to (optionally?)  specify a location when checking out an asset, as the primary (or default) location of the user is not always the location of the asset. Users can have two or three offices where they work and we would like to know which assets are located in which location. (In addition to the assets per user. This is a different use case, but for us, both would be good to have.)
So, we would basically like to have a similar function for checking _out_ an asset as we already have for checking back _in_.

**Comment by kgortmak** [comment]
Indeed.  Example:

User A is working for Business Unit 1 in location D, where he works 3 days a week, but also works 2 days every week at location Z. He owns a laptop. Users location therefore is Location D. When he works in location Z, he uses besides his laptop a dicking station and monitor provided by my colleague's in location Z.  When checking out, dock and monitor 'move' to location D, as being the users location. So they no longer turn up in inventory lists for location Z.  

I really think that it would be a great addition that you can, like with a checkIN, specify a different (actual) location for the asset than the user's location. That way you can still see the assets on the correct inventory, and the asset is still linked to the user. So when the user leaves and we have to check in his assets, I can communicate to my colleagues in Z that they can pickup the stuff from User A because he left.

**Comment by fracerutti** [comment]
+1 for this!
It would be very useful to have the option to manually set the asset location during checkout, just like it can be done now through bulk edit.

For example in our case, most users work remote most of the time and have "remote" location set up in their profile and therefore in their assets, but some of those users have monitors also assigned here in the office to use the days they work here.

**Comment by BETE-Landor** [comment]
+1 as well. We have users on-site and remote. Some on-site people have remote workstations. I want to be able to see where their particular hardware lives.

**Comment by TheYellowMango** [comment]
+1 for this feature. We have test stations. A user can check out items to setup at a test station. It will be great to be able to over-write the current location of the asset when the asset is being checked out to a user.

**Comment by technomooney** [comment]
i was going to make a new feature request. but i found this. i have users who have offices in different locations, i would like to set assets/accessories to have a current location and default location without relying on the location the users has on their profile.

**Comment by Marmachula** [comment]
I also need this functionality! We often land hardware for home office use, but the employees location remains set to the office. If I check out a device for HO use, I need to also update its location upon checkout!

> I know. And that is a cool feature. We need that, too. But in addition to that, it would be good for us to also be able to (optionally?) specify a location when checking out an asset, as the primary (or default) location of the user is not always the location of the asset. Users can have two or three offices where they work and we would like to know which assets are located in which location. (In addition to the assets per user. This is a different use case, but for us, both would be good to have.) So, we would basically like to have a similar function for checking _out_ an asset as we already have for checking back _in_.

**Comment by Marmachula** [comment]
How can we push this issue forward? It has been 7 years since it was created

**Comment by technomooney** [comment]
Considering there are 900+ issues, which I am sure are fairly high priority, like security or something, I can't imagine this is very high up on their priority list.... If I could help program it I would, but I don't know PHP.

**Comment by snipe** [comment]
This is, technologically, not a hard problem to solve. The problem is very much UX, as location inheriting from assigned location  (or falling back to default location) has been the behavior for a decade. Changing that now would certainly cause a lot of confusion.

**Comment by snipe** [comment]
More specifically, because you can change asset location via audit or bulk edit (both features I wrote that I hate), when the expectation is that the asset follows the assignee, how to we show that it was overridden, and how do we make that super super clear for admins not in the weeds every day? It just gets really confusing.

**Comment by technomooney** [comment]
Makes sense, I have to wonder if it would make sense to make it prefixed or suffixed with auto or use other indicators that would tell an admin that 1 it was auto assigned, and two where it was auto assigned from, maybe using a ruleset that could be setup by the admin? ¯\_(ツ)_/¯ I get that changes to a long used workflow is painful. But I don't know, I think having some things like this could be useful especially if it's related to multi LDAP / authentication systems. I think enabling something that's not oob and more an advanced feature toggle would be a good compromise
